stack/stack.toml

```diff
@@ -0,0 +1,85 @@
+id = "io.buildpacks.stacks.noble"
+homepage = "https://github.com/paketo-buildpacks/noble-base-stack"
+maintainer = "Paketo Buildpacks"
+
+platforms = ["linux/amd64", "linux/arm64"]
+
+[build]
+  description = "ubuntu:noble with compilers and shell utilities"
+  dockerfile = "./build.Dockerfile"
+  gid = 1000
+  shell = "/bin/bash"
+  uid = 1001
+
+  [build.args]
+    architecture = "amd64"
+
+    sources = """
+    deb http://archive.ubuntu.com/ubuntu noble main universe multiverse
+    deb http://archive.ubuntu.com/ubuntu noble-updates main universe multiverse
+    deb http://archive.ubuntu.com/ubuntu noble-security main universe multiverse
+    """
+
+    packages = """\
+    build-essential \
+    ca-certificates \
+    curl \
+    git \
+    jq \
+    libexpat1 \
+    libgmp-dev \
+    libssl3 \
+    libssl-dev \
+    libyaml-0-2 \
+    netbase \
+    openssl \
+    pkg-config \
+    tzdata \
+    xz-utils \
+    zlib1g-dev \
+    """
+
+    [build.platforms."linux/arm64".args]
+    architecture = "arm64"
+
+    sources = """
+    deb http://ports.ubuntu.com/ubuntu-ports/ noble main universe multiverse
+    deb http://ports.ubuntu.com/ubuntu-ports/ noble-updates main universe multiverse
+    deb http://ports.ubuntu.com/ubuntu-ports/ noble-security main universe multiverse
+    """
+
+[run]
+  description = "ubuntu:noble with some common dependencies like tzdata and openssl"
+  dockerfile = "./run.Dockerfile"
+  gid = 1000
+  shell = "/bin/bash"
+  uid = 1002
+
+  [run.args]
+    architecture = "amd64"
+
+    sources = """
+    deb http://archive.ubuntu.com/ubuntu noble main universe multiverse
+    deb http://archive.ubuntu.com/ubuntu noble-updates main universe multiverse
+    deb http://archive.ubuntu.com/ubuntu noble-security main universe multiverse
+    """
+
+    packages = """\
+    ca-certificates \
+    libexpat1 \
+    libssl3 \
+    libyaml-0-2 \
+    netbase \
+    openssl \
+    tzdata \
+    zlib1g \
+    """
+
+    [run.platforms."linux/arm64".args]
```

Comment:
Does this change above `enable` arm64 or `limit` it to arm64 versus the approach in the jammy stack where architecture is not metioned. More specifically I'm trying to understand if what's in jammy allows for every platform supported by ubuntu and this reduces that to just amd64 and arm64 or if not specifying in jammy defaults to only amd64

Reply:
> Does this change above enable arm64 or limit it to arm64 versus the approach in the jammy stack where architecture is not metioned.

This adds `arm64` based image as a result. The jammy base stack at the moment produces only `amd64`

> More specifically I'm trying to understand if what's in jammy allows for every platform supported by ubuntu and this reduces that to just amd64 and arm64 or if not specifying in jammy defaults to only amd64

The actual platforms are specified like that: https://github.com/paketo-buildpacks/jammy-base-stack/blob/e0ff3a8e1b4f767c7803d8c8f8d5162afe446116/stack/stack.toml#L5

Reply:
@pbusko thanks for the info